src/include/txn_inline.h

```diff
@@ -264,8 +264,10 @@ __wt_txn_op_delete_apply_prepare_state(WT_SESSION_IMPL *session, WT_REF *ref, bo
     page_del = ref->page_del;
     if (page_del != NULL) {
         page_del->timestamp = ts;
-        if (commit)
+        if (commit) {
             page_del->durable_timestamp = txn->durable_timestamp;
+            page_del->committed = true;
+        }
         WT_PUBLISH(page_del->prepare_state, prepare_state);
```

Comment:
I think this should be
```
 page_del->timestamp = ts;
 if (commit)
    page_del->durable_timestamp = txn->durable_timestamp;
 WT_PUBLISH(page_del->prepare_state, prepare_state);
 if (commit)
     WT_PUBLISH(page_del->committed, true);  
```

Reply:
Could you explain why? Is it important that other threads see the `prepare_state` update prior to the change to the `committed`?

That would be an unusual usage of WT_PUBLISH, which generally means "I've finished making changes to this data structure, now it's safe for other threads to see those changes"

Reply:
I think at least we should ensure we set the committed state after we have changed the prepared state. Otherwise, we may see a committed update without setting its prepared state properly.

Reply:
In addition, for a reader, it should see the page_del committed with the correct timestamps already set. If the compiler reorders to set committed to true then to set the timestamp, the reader may get wrong visibility.

Reply:
I think the commit is not fully done until we have set all the parameters so naturally to me setting the committed to true should be the last step in the commit.